File: ZoneSentry/ClientApp/src/api/models/RealtyServicePayment.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ApplicationUser } from './ApplicationUser';
import type { ConstructionCompany } from './ConstructionCompany';
import type { PaymentState } from './PaymentState';
import type { RealtyServiceOrder } from './RealtyServiceOrder';

export type RealtyServicePayment = {
    id?: number;
    payment?: number;
    paymentState?: PaymentState;
    dueDate?: string;
    periodStart?: string;
    periodEnd?: string;
    constructionCompany?: ConstructionCompany;
    orderer?: ApplicationUser;
    realtyServiceOrder?: RealtyServiceOrder;
};

File: ZoneSentry/Models/RealtyServicePayment.cs
namespace ZoneSentry.Models;

public class RealtyServicePayment
{
    public int Id { get; set; }
    
    public int Payment { get; set; }
    public PaymentState PaymentState { get; set; }
    public DateTime DueDate { get; set; }
    public DateTime PeriodStart { get; set; }
    public DateTime PeriodEnd { get; set; }
    
    public ConstructionCompany? ConstructionCompany { get; set; }
    public ApplicationUser? Orderer { get; set; }
    public RealtyServiceOrder? RealtyServiceOrder { get; set; }
}

public class RealtyServicePaymentDTO
{
    public int Id { get; set; }
    
    public int Payment { get; set; }
    public PaymentState PaymentState { get; set; }
    public DateTime DueDate { get; set; }
    public DateTime PeriodStart { get; set; }
    public DateTime PeriodEnd { get; set; }
    
    public ConstructionCompanyUserView? ConstructionCompany { get; set; }
    public ApplicationUserDTO? Orderer { get; set; }
    public RealtyServiceOrderDTO? RealtyServiceOrder { get; set; }
}